olga/analytics/models.py

```diff
@@ -149,3 +156,111 @@ def worlds_students_per_country_statistics(cls):  # pylint: disable=invalid-name
+        """
+        students_amount_percentage = format(
+            country_count_in_statistics / all_active_students_in_statistics * 100, '.2f'
+        )
+
+        if students_amount_percentage == '0.00':
+            students_amount_percentage = '~0'
+
+        return students_amount_percentage
+
+    @staticmethod
+    def is_country_exists(country):
+        """
+        Check if value is not a null.
+        """
+        if country != 'null':
+            return True
+
+    @staticmethod
+    def append_country_data_to_list(country_list, country, count, student_amount_percentage=None):
+        """
+        Make data for particular visualization format. Example for `datamap` or tabular view.
+        """
+        country_data = [str(pycountry.countries.get(alpha_2=country).alpha_3), count]
+
+        if student_amount_percentage is not None:
+            country_data += [student_amount_percentage]
+
+        country_list.append(country_data)
+
+    @classmethod
+    def create_worlds_students_per_country_data_formatted_to_render(cls, worlds_students_per_country):
+        # pylint: disable=invalid-name
+        """
+        Create convenient and necessary data formats to render it from view.
+
+        Graphs require list-format data.
+        """
+        datamap_format_countries_list = []
+        tabular_format_countries_list = []
+
+        all_active_students_in_statistics = sum(worlds_students_per_country.itervalues())
+        # pylint: disable=invalid-name
+
+        for country, count in worlds_students_per_country.iteritems():
+            student_amount_percentage = cls.get_student_amount_percentage(count, all_active_students_in_statistics)
+
+            if cls.is_country_exists(country):
+                cls.append_country_data_to_list(datamap_format_countries_list, country, count)
+
+                cls.append_country_data_to_list(
+                    tabular_format_countries_list, country, count, student_amount_percentage
+                )
+
+            else:
+                # Create students without country amount.
+                tabular_format_countries_list.append(('Unset', count, student_amount_percentage))
+
+        # Sort in descending order.
+        tabular_format_countries_list.sort(key=lambda row: row[1], reverse=True)
+
+        return datamap_format_countries_list, tabular_format_countries_list
+
+    @staticmethod
+    def get_countries_amount(tabular_format_countries_list):
+        """
+        Get countries amount in worlds students per country statistics as table.
+
+        If no countries append `Unset` field with zero countries, else get
+        """
+        if not tabular_format_countries_list:
+            tabular_format_countries_list.append(('Unset', 0, 0))
+
+            # Unset is not a country
+            countries_amount = 0
+
+        else:
+            # Delete unset country point from list
+            countries_amount = len(tabular_format_countries_list) - 1
```

Comment:
I did cool refactor for that, improved logic and cleared `get_countries_amount` method. Thanks.